internal/controller/cephfscg/cghandler.go

```diff
@@ -490,7 +490,7 @@ func DeleteTypedObjectList[T client.Object](ctx context.Context, k8sClient clien
 		// Ensure obj is a pointer
 		objCopy := obj
 
-		objPtr, ok := any(&objCopy).(client.Object)
+		objPtr, ok := any(objCopy).(client.Object)
```

Comment:
T is client.Object, and items is []T. Why do we need type assertion? This code will not compile if obj is not T.

The right fix is to remove lines 490-496

Reply:
That's the point and that's why lines 490-496 are there.

Reply:
Fixed as suggested by @nirs